File: 400.go
package leetcode

import "math"

func findNthDigit(n int) int {
	d := 1
	count := 9
	for {
		if tmp := d * count; tmp < n {
			n -= tmp
			d++
			count *= 10
		} else {
			break
		}
	}
	idx := (n - 1) / d
	bit := (n - 1) % d
	num := int(math.Pow10(d-1)) + idx
	return num / int(math.Pow10(d-1-bit)) % 10
}

File: 400.py
class Solution:
    def findNthDigit(self, n: int) -> int:
        x = 1
        while True:
            if (tmp := 9 * x * pow(10, x-1)) < n:
                n -= tmp
                x += 1
            else:
                idx = (n-1) // x
                bit = (n-1) % x
                num = pow(10, x-1) + idx
                # print(n, idx, num, bit)
                # return int(str(num)[bit])
                return num // pow(10, x - 1 - bit) % 10